StreamChatClient_v3/ChatClient_Tests.swift

```diff
@@ -368,6 +368,76 @@ class ChatClient_Tests: StressTestCase {
             Assert.willBeEqual(client.provideToken(), newUserToken)
         }
     }
+
+    func test_settingGuestUser() {
+        let client = Client(config: inMemoryStorageConfig, workerBuilders: [MessageSender.init], environment: testEnv.environment)
+        assert(testEnv.webSocketClient!.connect_calledCounter == 0)
+
+        let oldWSConnectEndpoint = client.webSocketClient.connectEndpoint
+
+        let newUserToken: Token = .unique
+        let newUserExtraData = NameAndImageExtraData(name: .unique, imageURL: .unique())
+        let newUser = GuestUser(id: .unique, extraData: newUserExtraData)
+
+        // Set up a new guest user
+        var setUserCompletionCalled = false
+        client.setGuestUser(userId: newUser.id, extraData: newUserExtraData, completion: { _ in setUserCompletionCalled = true })
+
+        AssertAsync {
+            // `WebSocketClient.disconnect(source:)` should be called once
+            Assert.willBeEqual(self.testEnv.webSocketClient?.disconnect_calledCounter, 1)
+
+            // Token should be flushed
+            Assert.willBeNil(client.provideToken())
+
+            // Database should be flushed
+            Assert.willBeTrue(self.testEnv.databaseContainer?.flush_called == true)
+
+            // New user id is set
+            Assert.willBeEqual(client.currentUserId, newUser.id)
+
+            // Make sure `guest` endpoint is called
+            Assert.willBeEqual(self.testEnv.apiClient!.request_endpoint,
+                               AnyEndpoint(.token(userId: newUser.id, extraData: newUserExtraData)))
+
+            // Make sure the completion is not called yet
+            Assert.willBeFalse(setUserCompletionCalled)
```

Comment:
This should probably be `Assert.staysFalse` or you can move it after the `AsertAsync` block and do just `XCTAssertFalse(setUserCompletionCalled)`. The `AssertAsync` blocks the execution until all conditions are met.

Reply:
fixed in c2f9b54ca1e776672e3feafe43f9cca993534e08